services/QuillLMS/app/controllers/blog_posts_controller.rb

```diff
@@ -19,22 +24,35 @@ def student_center_index
     topic_names.each do |name|
       @topics.push({ name: name, slug: CGI::escape(name.downcase.gsub(' ','-'))})
     end
-    @blog_posts = BlogPost.where(draft: false, topic: topic_names)
+    @blog_posts = BlogPost.for_topics(topic_names)
     render :index
   end
 
   def show
-    find_by_hash = { slug: params[:slug] }
-    find_by_hash[:draft] = false unless @role == 'staff'
-    @blog_post = BlogPost.find_by!(find_by_hash)
-    @topic = @blog_post.topic
-    @display_paywall = true unless @blog_post.can_be_accessed_by(current_user)
-    @blog_post.increment_read_count
-    @author = @blog_post.author
-    @most_recent_posts = BlogPost.where("draft = false AND id != #{@blog_post.id}").order('updated_at DESC').limit(3)
-    @title = @blog_post.title
-    @description = @blog_post.subtitle || @title
-    @image_link = @blog_post.image_link
+    draft_statuses = @role == 'staff' ? [true, false] : false
```

Comment:
I know this wasn't in the older version of this method, but it probably makes sense to use `User::STAFF` here.

Reply:
Yeah, good call. I'll refactor this to be better.